### ISSUE TITLE
Open connection with pre configured session

### ISSUE THREAD

**Issue by Harm10** [Altran-PT-GDC/Robot-Framework-Mainframe-3270-Library#76]
With the wc3270 wizard you can create pre configured sessions. Can these sessions be used in some way with Open Connection?

**Comment by Harm10** [comment]
I have removed the part about extra_args to create a separate issue for this as there are more things to clear (for example documentation for usage of extra_args seems to be wrong when using wc3270 on Windows).

**Comment by robinmatz** [comment]
Hi @Harm10 ,

I have been experimenting with a new keyword, `Open Connection From Session File`. You  can check out the code from the [open-connection-from-session-file branc](https://github.com/robinmatz/Robot-Framework-Mainframe-3270-Library/tree/open-connection-from-session-file) on my fork.

Maybe you can give me some feedback on it already.

**Comment by robinmatz** [comment]
You can clone my repository and check out the branch.

**Comment by Harm10** [comment]
So I did and after starting RIDE I suddenly get several local updates:
![afbeelding](https://user-images.githubusercontent.com/1517188/224547070-7c211b1d-30b9-4b87-ae06-466154264fcd.png)
For example the change in SOURCES.txt:
![afbeelding](https://user-images.githubusercontent.com/1517188/224547124-129874c3-2c2e-4f38-9c35-79d45942e5d6.png)
Where do these changes come from?

**Comment by Harm10** [comment]
I probably did something wrong. I cloned your repo and in VSCode went to the branch you are working on. Then I wanted to push a small text update but I am not allowed to do that. What am I doing wrong?
The text update is to point users to the session wizard for creating session files (much easier!).

The first test with one of my session files went fine (correct size of windows console!).
Looking into that session file the way the host is specified is different than in your example:
`wc3270.hostname: pub400.com`
The port is not specified like this but is in the block that is read when building the session (according to the comments in session file itself). I am using wc3270 v4.2ga8

**Comment by robinmatz** [comment]
> I probably did something wrong. I cloned your repo and in VSCode went to the branch you are working on. Then I wanted to push a small text update but I am not allowed to do that. What am I doing wrong?

You did nothing wrong. One cannot push changes to another person's repository if they are not a collaborator.
If it's just a small change, you can write your suggestions here.

> The text update is to point users to the session wizard for creating session files (much easier!).

Sure, but also keep in mind that this project supports four types of emulators depending on the operating system and visibility settings. At least for x3270 and s3270 there is no session wizard.

> The first test with one of my session files went fine but the windows console dimensions are always the same irrespective of the model chosen while the session file when used directly does open a bigger window.

For simplicity, I would like you to leave out the aspect of different models. We can continue the discussion on this in #31.

> Looking into that session file the way the host is specified is different than in your example:
wc3270.hostname: pub400.com

You can actually do both `*hostname: pub400.com` or `wc3270.hostname: pub400.com`. They are both valid options
![image](https://user-images.githubusercontent.com/76647407/224610761-17f4892d-1ec3-41df-84e4-ca6b92e22900.png)
(https://x3270.miraheze.org/wiki/Session_file)

> The port is not specified like this but is in the block that is read when building the session (according to the comments in session file itself).

Sorry, I don't get what you mean. Can you give me an example? I wrote a test called `Test Connection From Session File` in atest/connection.robot that specifies the port with `*port: 992` and it passes just fine.

**Comment by Harm10** [comment]
I compared 2 files with the only difference on the port. I thought there is no port code there but it is set up differently.
With port 23 there is nothing there and with port 999 the line becomes
`wc3270.hostname: pub400.com:999`

I did not know there is no session wizard for x3270/s3270. I still think you should make some reference to this syntax. Or is this only for wc3270?

You did not respond on my differences for example in SOURCES.txt As I cannot change your fork this probably is irrelevant now?

**Comment by robinmatz** [comment]
Ok, this seems to be an alternative way of specifying the port.

Have you read the documentation I provided for the Open Connection From Session File keyword? I provided many links to the x3270 documentation on session files, including the syntax.

**Comment by Harm10** [comment]
I looked again at [https://x3270.miraheze.org/wiki/Session_file](url) and saw that the session wizard is also mentioned there. So I guees having this link in the description should be enough.

Will this update be included in version 3.1?

**Comment by robinmatz** [comment]
@samuelpcabral , this feature is not yet included in 3.1. So there is no need to update the documentation

**Comment by Harm10** [comment]
@robinmatz Will it then be included in 3.2?

**Comment by robinmatz** [comment]
@Harm10 yes, it will be. I am actually curious about your suggestion on the documentation on the new keyword. Could you paste them in this conversation?

**Comment by Harm10** [comment]
@robinmatz First about the text suggestion I had this `The easiest way to create a session file is to use the wizard found in your wc3270 installation folder: wc3270wiz.exe`
But after that I mentioned on Slack that I did not know this wizard only exists for wc3270. So whether this addition should be added I am not sure any more.

What I would suggest for the documentation is an example like this:
`wc3270.hostname: pub400.com:23`
`wc3270.model: 2`

As you can see the model number is part of the session file generated by the wizard. So if you only want to support model 2 at the moment you can easily check what is in the session file supplied. If no model is there you can also easily add it in a copy of the file to use to circumvent the default of 4 (which I find peculiar as most mainframes still use model 2 I think).
Another option could be that this package contains a sample session file that users can use?

I am also searching for a way to test other models. At work we use Aviva emulator but it makes no difference at all which kind of model in the session file is specified the screen is always the same (24x80).
The only thing I notice is that when you change the font size the window automatically changes depending on the font size.

Speaking about the font size. The wizards writes this in the code block below the regular lines. I have no idea how you can define it with a "normal" line. I can confirm that a bigger font size works fine from that code block.

I would also like to add that the wizard does not allow prepending the host (see [http://x3270.bgp.nu/Windows/wc3270-man.html#Description](url)). That part of the documentation does not even describe that you can also use prepend C: meaning that the started session will be a CICS session which has somewhat different capabilities. Using such a prepend in the regular Open Connection works fine.

Is this enough input? :-)

**Comment by robinmatz** [comment]
Hi @Harm10, 
sorry that I failed to reply to your comment until now. I tried to work in your suggestions as much as possible. I opted not to make the documentation to specific to wc3270/the session wizzard. As you mentioned, this is already described in the links to the x3270 wiki. However, I did include the examples you provided in the keyword documentation.

As for the model, I opted to just support model 2 at the moment, because I wanted to reduce the complexity for now. However, users are adviced to specify model 2 in the keyword documentation, and the library will check if model 2 was specified. If another model is specified, there will be an error message.

I did not include the comment about the wizard not allowing host prefixes (`Y:`, `L:`, `C:`, etc.) because, again, this is quite specific to the wizard.

I created a MR #96 and I hope that the next version including the changes will be released soon.
